lib/stagehand/stagehand.coffee

```diff
@@ -0,0 +1,86 @@
+spawn = require('child_process').spawn
+SdkService = require '../sdk/sdk_service'
+Picker = require '../ui/picker'
+path = require 'path'
+
+# Stagehand support
+class Stagehand
+  @activate: =>
+    return new Promise (resolve, reject) =>
+      cmd  = SdkService.getCommandPath 'pub'
+      args = ['global', 'activate', 'stagehand']
+      errors = ''
+
+      process = spawn cmd, args
+      process.stderr.on 'data', (data) =>
+        errors = errors + data.toString()
+      process.on 'exit', (code) =>
+        if code is 0
+          resolve()
+        else
+          reject(errors)
+
+  @getProjectTemplates: =>
+    return new Promise (resolve, reject) =>
+      cmd  = SdkService.getCommandPath 'pub'
+      args = ['global', 'run', 'stagehand', '--machine']
+      errors = ''
+      json = ''
+
+      process = spawn cmd, args
+      process.stdout.on 'data', (data) =>
+        json = json + data.toString()
+      process.stderr.on 'data', (data) =>
+        errors = errors + data.toString()
+      process.on 'exit', (code) =>
+        if code is 0
+          resolve(JSON.parse(json))
+        else
+          reject(errors)
+
+  @showProjectTemplates: =>
+    picker = new Picker
+    @getProjectTemplates().then (projectTemplates) =>
+      items = projectTemplates.map (pt) =>
+        {item: pt, displayName: pt.label, description: pt.description}
+
+      return picker.selectFrom(items)
+
+  @generate: (projectTemplate) =>
+    picker = new Picker
+    projectPaths  = atom.project.getPaths()
+    generateProject = (projectPath) =>
+      atom.notifications.addInfo "[dart-tools] Generating #{projectTemplate.label}."
+
+      cmd  = SdkService.getCommandPath 'pub'
+      args = ['global', 'run', 'stagehand', projectTemplate.name]
+      errors = ''
+      output = ''
+
+      process = spawn cmd, args,
+        cwd: projectPath
+      process.stdout.on 'data', (data) =>
+        output = output + data.toString()
+      process.stderr.on 'data', (data) =>
+        errors = errors + data.toString()
+      process.on 'exit', (code) =>
+        if code is 0
+          atom.notifications.addInfo "[dart-tools] Generated #{projectTemplate.label}!"
+        else
+          atom.notifications.addError "[dart-tools] Failed to generate #{projectTemplate.label}!",
+            detail: errors or output
+
+    if projectPaths.length is 0
+      atom.notifications.addError "[dart-tools] Cannot run Stagehand without a project directory open."
```

Comment:
>  Are we requiring there to be a project open to run stagehand?

In Atom a project is just a set of directories that is open inside of Atom, so we're requiring the user have at least one directory open in the project before continuing. By default if you open Atom, it's rooted in the home directory, so if you asked a user to type a relative path to the location to run Stagehand it's very possible they might fatfinger "Projects/dart/cool-new-webapp" into "Proects/dart/cool-new-webapp", so I thought it would be safer to leave the directory creation to something with tab-complete like Advanced New File, or just doing it through the Atom GUI.